### PR TITLE
feat: use stacks schema merger

### PR DESCRIPTION
**Pull request by ansgarm**
Merge all the schemas (for stacks stack files) 💯

This brings support for completions for `inputs` and `providers` in `component` blocks and for showing descriptions on hover among other schema related features. Note: this currently only supports local components with a source in a relative directory.

### CODE REVIEW

File: internal/features/stacks/state/stack_store.go [hunk 1]
```diff
@@ -306,6 +308,11 @@ func (s *StackStore) UpdateMetadata(path string, meta *tfstack.Meta, mErr error)
 		return err
 	}
 
+	err = txn.Insert(s.tableName, record)
+	if err != nil {
+		return err
+	}
+
```
Comment (dbanck) [inline internal/features/stacks/state/stack_store.go:315]:
```suggestion
```

It was so nice, we inserted it twice

File: internal/features/stacks/events.go [hunk 2]
```diff
@@ -205,34 +205,38 @@ func (f *StacksFeature) decodeStack(ctx context.Context, dir document.DirHandle,
 			}
 
 			spawnedIds, err := loadStackComponentSources(ctx, f.store, f.bus, path)
+			if err != nil {
+				return spawnedIds, err
```
Comment (dbanck) [inline internal/features/stacks/events.go:209]: Do we need to return here? I think we can still try to load the schemas, even if decoding the components fails

File: internal/features/stacks/events.go [hunk 3]
```diff
@@ -205,34 +205,38 @@ func (f *StacksFeature) decodeStack(ctx context.Context, dir document.DirHandle,
...
+			}
+			spawnedIds = append(spawnedIds, eSchemaId)
 
-	eSchemaId, err := f.stateStore.JobStore.EnqueueJob(ctx, job.Job{
-		Dir: dir,
-		Func: func(ctx context.Context) error {
-			return jobs.PreloadEmbeddedSchema(ctx, f.logger, schemas.FS,
-				f.store, f.stateStore.ProviderSchemas, path)
+			return spawnedIds, nil
```
Comment (dbanck) [inline internal/features/stacks/events.go:233]: What do you think about adding a `deferIds := make(job.IDs, 0)` to the top of this function, appending to it, and returning it here? I think that would make it more readable.

File: internal/features/stacks/decoder/path_reader.go [hunk 4]
```diff
@@ -35,28 +48,46 @@ func (pr *PathReader) PathContext(path lang.Path) (*decoder.PathContext, error)
...
 	}
 
 	schema, err := stackschema.CoreStackSchemaForVersion(version)
 	if err != nil {
 		return nil, err
 	}
 
+	sm := stackschema.NewStackSchemaMerger(schema)
+	sm.SetStateReader(stateReader)
+	sm.SetModuleReader(moduleReader)
```
Comment (dbanck) [inline internal/features/stacks/decoder/path_reader.go:73]: In line with the [comment on terraform-schema](https://github.com/hashicorp/terraform-schema/pull/376#discussion_r1695506664), what do you think about having only one reader in the schema merger?

**Review by dbanck — COMMENTED** [review]
Nice work! I left some suggestions

File: internal/features/stacks/events.go (hunk 2 above)
Comment (ansgarm) [reply internal/features/stacks/events.go]: Would you then just ignore the error or keep it for later? Which error would you return if loading the schemas returns one as well?

Comment (dbanck) [reply internal/features/stacks/events.go]: I would log the error from `loadStackComponentSources` and proceed with loading the schemas.

If scheduling the schema job fails, something must really be broken, so I think returning there makes sense.

**Review by dbanck — APPROVED** [review]
Thanks for addressing the feedback

**Pull request merged**